sktime/classification/model_selection/_tune.py

```diff
@@ -0,0 +1,404 @@
+
+        self.estimator_ = GridSearchCV(**gcsvargs)
+
+    def _fit(self, X, y):
+        """Fit time series classifier to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            if self.get_tag("X_inner_mtype") = "numpy3D":
+                3D np.ndarray of shape = [n_instances, n_dimensions, series_length]
+            if self.get_tag("X_inner_mtype") = "nested_univ":
+                pd.DataFrame with each column a dimension, each cell a pd.Series
+            for list of other mtypes, see datatypes.SCITYPE_REGISTER
+            for specifications, see examples/AA_datatypes_and_datasets.ipynb
+        y : 1D np.array of int, of shape [n_instances] - class labels for fitting
+            indices correspond to instance indices in X
+
+        Returns
+        -------
+        self : Reference to self.
+        """
+        estimator = self._get_delegate()
+        estimator.fit(X=X, y=y)
+
+        fitted_param_names = [
+            "cv_results_",
+            "best_estimator_",
+            "best_score_",
+            "best_params_",
+            "best_index_",
+            "scorer_",
+            "n_splits_",
+            "refit_time_",
+            "multimetric_",
+            "classes_",
+        ]
+
+        for p in fitted_param_names:
+            if hasattr(estimator, p):
+                val = getattr(estimator, p)
+                setattr(self, p, val)
+
+        return self
+
+    def _predict_proba(self, X):
+        """Predicts labels probabilities for sequences in X.
+
+        private _predict_proba containing the core logic, called from predict_proba
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            if self.get_tag("X_inner_mtype") = "numpy3D":
+                3D np.ndarray of shape = [n_instances, n_dimensions, series_length]
+            if self.get_tag("X_inner_mtype") = "nested_univ":
+                pd.DataFrame with each column a dimension, each cell a pd.Series
+            for list of other mtypes, see datatypes.SCITYPE_REGISTER
+            for specifications, see examples/AA_datatypes_and_datasets.ipynb
+
+        Returns
+        -------
+        y : 2D array of shape [n_instances, n_classes] - predicted class probabilities
+            1st dimension indices correspond to instance indices in X
+            2nd dimension indices correspond to possible labels (integers)
+            (i, j)-th entry is predictive probability that i-th instance is of class j
+        """
+        if not self.refit:
```

Comment:
@fkiraly Is there a reason we're directly calling the base estimator's method and not estimator's? (`return self.estimator._predict_proba(X)`)

Reply:
if `refit=False`, then the `sklearn` grid search (see the same parameter there) does not compute probabilistic predictions and will raise an exception. In this case, we instead need to carry out the default logic in `BaseClassifier`, which lives in `_predict_proba`.

Or, `predict_proba`, there may be a minor issue in calling the private, not the public method, but I must have thought the private one is ok.

Reply:
Let me rephrase my question, let's say that `refit=False`. In this case, if we were to call `predict()` (or `predict_proba()`) on it, shouldn't it simply call the underlying estimator's `predict` (or `predict_proba()`)? In sklearn's implementation, `predict()` only happens when `refit=True` so we're already diverging from the expected contract - we at least need to add more documentation if we're making a conscious decision to diverge.

Reply:
apologies for the delay in responding. pydata, travels etc.

Re predict dispatch, I'm not sure - as said, if `refit=False`, a call to `sklearn`  `predict_proba` will raise an exception, so we have to do sth else.

A call to `predict` will dispatch to the underlying `predict`, as you suggest.

So, is that not what you think should happen? Maybe let's discuss in one of the dev meet-ups.

Reply:
from discussion with @achieveordie on discord: @achieveordie was right, `refit=False` impacts not just `predict_proba` but also `predict` - it was my mistake and confusion to think it only affects `predict_proba`.

Solved now by not testing `refit=False`, as that will violate the classifier contract (non-functional `predict`)